src/topology/opens.lean

```diff
@@ -148,33 +143,60 @@ begin
     rw is_basis_iff_nbhd,
     intros U x hx,
     rcases h U with ⟨Us, hUs, H⟩,
-    replace H := congr_arg subtype.val H,
+    replace H := congr_arg (coe : _ → set α) H,
     rw Sup_s at H,
-    change x ∈ U.val at hx,
+    change x ∈ ↑U at hx,
     rw H at hx,
     rcases set.mem_sUnion.mp hx with ⟨sV, ⟨⟨V, H₁, H₂⟩, hsV⟩⟩,
     refine ⟨V,hUs H₁,_⟩,
     cases V with V hV,
     dsimp at H₂, subst H₂,
     refine ⟨hsV,_⟩,
-    change V ⊆ U.val, rw H,
+    change V ⊆ U, rw H,
     exact set.subset_sUnion_of_mem ⟨⟨V, _⟩, ⟨H₁, rfl⟩⟩ }
 end
 
+/-- The preimage of an open set, as an open set. -/
+def comap {f : α → β} (hf : continuous f) (V : opens β) : opens α :=
+⟨f ⁻¹' V.1, hf V.1 V.2⟩
+
+@[simp] lemma comap_id (U : opens α) : U.comap continuous_id = U := by { ext, refl }
+
+lemma comap_mono {f : α → β} (hf : continuous f) {V W : opens β} (hVW : V ⊆ W) :
+  V.comap hf ⊆ W.comap hf :=
+λ _ h, hVW h
+
+@[simp] lemma coe_comap {f : α → β} (hf : continuous f) (U : opens β) :
+  ↑(U.comap hf) = f ⁻¹' U := rfl
+
+@[simp] lemma comap_val {f : α → β} (hf : continuous f) (U : opens β) :
+  (U.comap hf).1 = f ⁻¹' U.1 := rfl
+
+protected lemma comap_compose {g : β → γ} {f : α → β} (hg : continuous g) (hf : continuous f)
+  (U : opens γ) : U.comap (hg.comp hf) = (U.comap hg).comap hf :=
+by { ext1, simp only [coe_comap, preimage_preimage] }
+
+/-- A homeomorphism induces an equivalence on open sets, by taking comaps. -/
+@[simp] protected def equiv (f : α ≃ₜ β) : opens α ≃ opens β :=
+{ to_fun := opens.comap f.symm.continuous,
+  inv_fun := opens.comap f.continuous,
+  left_inv := by { intro U, ext1,
+    simp only [coe_comap, ← preimage_comp, f.symm_comp_self, preimage_id] },
+  right_inv := by { intro U, ext1,
+    simp only [coe_comap, ← preimage_comp, f.self_comp_symm, preimage_id] } }
+
 end opens
 
+/-- The open neighborhoods of a point. See also `opens` or `nhds`. -/
+def open_nhds_of (x : α) : Type* := { s : set α // is_open s ∧ x ∈ s }
```

Comment:
I would expect some compatibility lemmas with `opens` and `nhds`... is that planned for a future PR?

Reply:
I don't have anything lined up. I define a function that takes an argument of this type (and to prove properties it was easiest to work with this subtype, not just a separate hypothesis), but I didn't need any operations on this type.